sdk/storage/azure-storage-common-data-movement/src/main/java/com/azure/storage/common/data/movement/PathScanner.java

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.storage.common.data.movement;
+
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+import reactor.core.scheduler.Schedulers;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.*;
+import java.util.stream.BaseStream;
+import java.util.stream.Stream;
+
+public class PathScanner {
+    private final String basePath;
+    private final boolean isDir;
```

Comment:
Do we need to pass in isDir? We could use Files.isDirectory in the code below instead of a boolean

Reply:
+1

Reply:
btw. what does it mean to scan a file?

Reply:
I'll move this then, but in that case the Factory becomes pretty bare, since it only stores path and the build method would just call the Scanner's constructor. Is it even necessary then?

Scanning a file just returns a flux with that file alone; I was thinking that it's useful to handle as it would mean scanning is more streamlined. Rather than a user/caller needing to figure out if its file or folder and whether or not to make a scanner, they'd just need to pass their path on the scanner and handle the Flux resulting uniformly.